src/libraries/Microsoft.PowerFx.Json/Public/Types/UserProvidedTypeDefinitions/FormulaTypeSchema.cs

```diff
@@ -56,15 +62,16 @@ public FormulaType ToFormulaType(DefinedTypeSymbolTable definedTypeSymbols, Form
             }
 
             var logicalNameToRecordType = settings.LogicalNameToRecordType;
-            if (typeName == SchemaTypeName.ExpandableTableTypeName.Name && logicalNameToRecordType != null)
-            {
-                return logicalNameToRecordType.Invoke(this.Description).ToTable();
-            }
-            else if (typeName == SchemaTypeName.ExpandableRecordTypeName.Name && logicalNameToRecordType != null)
+            if ((typeName == SchemaTypeName.CustomTableTypeName.Name || typeName == SchemaTypeName.CustomRecordTypeName.Name) && logicalNameToRecordType != null)
             {
-                return logicalNameToRecordType.Invoke(this.Description);
+                if (Type.IsTable)
+                {
```

Comment:
nit,  you could factor the shared code, like:

var recordType = logicalNameToRecordType.Invoke(this.CustomTypeName);

if (Type.IsTable) {
  return recordType.ToTable();
}
else {
return recordType;
}